pcsx2/gui/MainFrame.cpp

```diff
@@ -636,35 +638,49 @@ void MainEmuFrame::ApplyCoreStatus()
 		}
 	}
 
+	const u8 TotalSources = static_cast<u8>(CDVD_SourceType::NumberOfSources);
+	const CDVD_SourceType Source = g_Conf->CdvdSource;
+	const wxChar* FullBootHelpText[TotalSources];
+	// (X axis - reboot/boot,  Y axis - full/fast,  Z axis - source medium)
+	const wxChar* MenuItemLabels[2][2][TotalSources] = {};
+
+	FullBootHelpText[enum_cast(CDVD_SourceType::Iso)] = _("Boot the VM using the current ISO source media");
+	FullBootHelpText[enum_cast(CDVD_SourceType::Plugin)] = _("Boot the VM using the current CD/DVD source media");
+	FullBootHelpText[enum_cast(CDVD_SourceType::NoDisc)] = _("Boot the VM without any source media");
+
+	MenuItemLabels[0][0][enum_cast(CDVD_SourceType::Iso)]    = _("Boot ISO (f&ull)");
+	MenuItemLabels[0][1][enum_cast(CDVD_SourceType::Iso)]    = _("Boot ISO (&fast)");
+	MenuItemLabels[1][0][enum_cast(CDVD_SourceType::Iso)]    = _("Reboot ISO (f&ull)");
+	MenuItemLabels[1][1][enum_cast(CDVD_SourceType::Iso)]    = _("Reboot ISO (&fast)");
+	MenuItemLabels[0][0][enum_cast(CDVD_SourceType::Plugin)] = _("Boot CDVD (f&ull)");
+	MenuItemLabels[0][1][enum_cast(CDVD_SourceType::Plugin)] = _("Boot CDVD (&fast)");
+	MenuItemLabels[1][0][enum_cast(CDVD_SourceType::Plugin)] = _("Reboot CDVD (f&ull)");
+	MenuItemLabels[1][1][enum_cast(CDVD_SourceType::Plugin)] = _("Reboot CDVD (&fast)");
+	MenuItemLabels[0][0][enum_cast(CDVD_SourceType::NoDisc)] = _("Boot BIOS");
+	MenuItemLabels[1][0][enum_cast(CDVD_SourceType::NoDisc)] = _("Reboot BIOS");
+
+
 	if( cdvd )
 	{
-		if( vm )
-		{
-			cdvd->SetItemLabel(_("Reboot CDVD (f&ull)"));
+		cdvd->SetItemLabel(MenuItemLabels[ActiveVM][0][enum_cast(Source)]);
+		if( ActiveVM )
 			cdvd->SetHelp(_("Hard reset of the active VM."));
-		}
 		else
-		{
-			cdvd->SetItemLabel(_("Boot CDVD (f&ull)"));
-			cdvd->SetHelp(_("Boot the VM using the current DVD or Iso source media"));
-		}	
+			cdvd->SetHelp(FullBootHelpText[enum_cast(Source)]);
 	}
 
-	if( cdvd2 )
-	{
-		if( vm )
-		{
-			cdvd2->SetItemLabel(_("Reboot CDVD (&fast)"));
-			cdvd2->SetHelp(_("Reboot using fast BOOT (skips splash screens)"));
-		}
-		else
-		{
-			cdvd2->SetItemLabel(_("Boot CDVD (&fast)"));
-			cdvd2->SetHelp(_("Use fast boot to skip PS2 startup and splash screens"));
-		}
-	}
+	if(!cdvd2)
+		cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2, MenuItemLabels[ActiveVM][1][enum_cast(Source)]);
```

Comment:
It really isn't so difficult. This:
```
wxMenuItem *cdvd_fast = menubar.FindItem(MenuId_Boot_CDVD2);
if (g_Conf->CdvdSource == CDVD_SourceType::NoDisc) {
	if (cdvd_fast)
		m_menuSys.Destroy(cdvd_fast);
} else {
	wxString label;
	wxString help_text;
	if (ActiveVM) {
		if (g_Conf->CdvdSource == CDVD_SourceType::Iso)
			label = _("Reboot ISO (&fast)");
		else
			label = _("Reboot CDVD (&fast)");
		help_text = _("Reboot using fast boot (skips splash screens)");
	} else {
		if (g_Conf->CdvdSource == CDVD_SourceType::Iso)
			label = _("Boot ISO (&fast)");
		else
			label = _("Boot CDVD (&fast)");
		help_text = _("Use fast boot to skip PS2 startup and splash screens");
	}

	if (cdvd_fast == nullptr) {
		m_menuSys.Insert(1, MenuId_Boot_CDVD2, label, help_text);
	} else {
		cdvd_fast->SetItemLabel(label);
		cdvd_fast->SetHelp(help_text);
	}
}
```
works for me. I didn't need to create and destroy an item that doesn't get used. I also didn't need to use an array and spam casts.